testcontainers-dapr/src/test/java/io/dapr/testcontainers/DaprContainerTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.testcontainers;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.domain.Metadata;
+import io.dapr.client.domain.State;
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.Testcontainers;
+
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.configureFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static java.util.Collections.singletonMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DaprContainerTest {
```

Comment:
This test fits more as an integration test. Unit tests should not depend on Docker, network or any IO.

Reply:
I totally agree, but this test is not testing the Dapr APIs or anything deeper in Dapr, like for example: https://github.com/dapr/java-sdk/blob/master/sdk-tests/src/test/java/io/dapr/it/state/HelloWorldClientIT.java 

This one is testing the testcontainers mechanism to start the Dapr sidecar configured in the right way to work for certain use cases. 

In the future, I would love to move the IT test to all use this mechanism. But right now it doesn't make any sense for me to move this test to `sdk-tests` -> `io/dapr/it`.

Reply:
Let's discuss this because I see it in a different way. Moving to IT has nothing to do with using Dapr or not, but to the fact that it has "unmocked" dependencies.

Reply:
moved to sdk-tests